packages/fuel-indexer/src/executor.rs

```diff
@@ -488,22 +491,16 @@ impl Executor for WasmIndexExecutor {
         let ptr = arg.get_ptr();
         let len = arg.get_len();
 
-        let res = spawn_blocking(move || fun.call(ptr, len)).await?;
+        let res = timeout(
+            Duration::from_secs(self.timeout),
+            spawn_blocking(move || fun.call(ptr, len)),
+        )
+        .await;
 
         if let Err(e) = res {
-            error!("WasmIndexExecutor handle_events failed: {}.", e.message());
-            let frames = e.trace();
-            for (i, frame) in frames.iter().enumerate() {
-                println!(
-                    "Frame #{}: {:?}::{:?}",
-                    i,
-                    frame.module_name(),
-                    frame.function_name()
-                );
-            }
-
+            error!("WasmIndexExecutor handle_events failed: {e:?}.");
             self.db.lock().await.revert_transaction().await?;
-            return Err(IndexerError::RuntimeError(e));
+            return Err(IndexerError::from(e));
```

Comment:
Could we potentially add a `match` here to allow for us to inform an operator that something timed out as opposed to a different error?

On the other hand, they'll get that information this way as well. A matter of preference, I suppose.